bundle/direct/dresources/dashboard.go

```diff
@@ -0,0 +1,316 @@
+package dresources
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"path"
+	"slices"
+	"strings"
+
+	"github.com/databricks/cli/bundle/config/resources"
+	"github.com/databricks/cli/bundle/deployplan"
+	"github.com/databricks/databricks-sdk-go"
+	"github.com/databricks/databricks-sdk-go/apierr"
+	"github.com/databricks/databricks-sdk-go/service/dashboards"
+	"golang.org/x/sync/errgroup"
+)
+
+// Terraform implementation reference: https://github.com/databricks/terraform-provider-databricks/blob/main/dashboards/resource_dashboard.go
+type ResourceDashboard struct {
+	client *databricks.WorkspaceClient
+}
+
+func (*ResourceDashboard) New(client *databricks.WorkspaceClient) *ResourceDashboard {
+	return &ResourceDashboard{client: client}
+}
+
+func (*ResourceDashboard) PrepareState(input *resources.Dashboard) *resources.DashboardConfig {
+	return &input.DashboardConfig
+}
+
+func (r *ResourceDashboard) RemapState(state *resources.DashboardConfig) *resources.DashboardConfig {
+	dashboard := &resources.DashboardConfig{
+		Dashboard: dashboards.Dashboard{
+			DisplayName: state.DisplayName,
+			Etag:        state.Etag,
+			ParentPath:  state.ParentPath,
+			WarehouseId: state.WarehouseId,
+			ForceSendFields: filterFields[dashboards.Dashboard](state.ForceSendFields, []string{
+				"CreateTime",
+				"DashboardId",
+				"LifecycleState",
+				"Path",
+				"UpdateTime",
+				"SerializedDashboard",
+			}...),
+
+			// Clear output only fields. They should not show up on remote diff computation.
+			CreateTime:     "",
+			DashboardId:    "",
+			LifecycleState: dashboards.LifecycleState(""),
+			Path:           "",
+			UpdateTime:     "",
+
+			// Serialized dashboard is ignored for remote diff changes.
+			// They are only relevant for local diff changes.
+			SerializedDashboard: "",
+		},
+
+		EmbedCredentials: state.EmbedCredentials,
+		ForceSendFields: filterFields[resources.DashboardConfig](state.ForceSendFields, []string{
+			"SerializedDashboard",
+		}...),
+
+		// Serialized dashboard is ignored for remote diff changes.
+		// They are only relevant for local diff changes.
+		SerializedDashboard: "",
+	}
+
+	return dashboard
+}
+
+func (r *ResourceDashboard) DoRefresh(ctx context.Context, id string) (*resources.DashboardConfig, error) {
+	var dashboard *dashboards.Dashboard
+	var publishedDashboard *dashboards.PublishedDashboard
+
+	g, ctx := errgroup.WithContext(ctx)
+
+	g.Go(func() error {
+		var err error
+		dashboard, err = r.client.Lakeview.Get(ctx, dashboards.GetDashboardRequest{
+			DashboardId: id,
+		})
+		if err != nil {
+			return fmt.Errorf("failed to get draft dashboard: %w", err)
```

Comment:
nit: we already have context about the error something like, "Error: cannot update resources.dashboard.foo", so prefixes don't need to says that it is "failed". Could be shorter: "reading dashboard", "reading published status". In this case, even that could be omitted since it's pretty clear what's happening from API endpoint name.